src/transformers/models/deformable_detr/modeling_deformable_detr.py

```diff
@@ -2172,6 +2160,39 @@ def loss_boxes(self, outputs, targets, indices, num_boxes):
         losses["loss_giou"] = loss_giou.sum() / num_boxes
         return losses
 
+    def loss_masks(self, outputs, targets, indices, num_boxes):
+        """
+        Compute the losses related to the masks: the focal loss and the dice loss.
+
+        Targets dicts must contain the key "masks" containing a tensor of dim [nb_target_boxes, h, w].
+        """
+        if "pred_masks" not in outputs:
+            raise KeyError("No predicted masks found in outputs")
+
+        source_idx = self._get_source_permutation_idx(indices)
+        target_idx = self._get_target_permutation_idx(indices)
+        source_masks = outputs["pred_masks"]
+        source_masks = source_masks[source_idx]
+        masks = [t["masks"] for t in targets]
+        # TODO use valid to mask invalid areas due to padding in loss
```

Comment:
Is this left-over from this original codebase or will the invalid area computation be added?

Reply:
This is a leftover from the original codebase.